url647b55da7c_Nikhilkumar13_emuSr_tgz-pJ8-MainJ8/src/backend/Memory.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package backend;
+
+import frontend.FrontEnd;
+import java.awt.Color;
+import java.util.AbstractCollection;
+import java.util.HashMap;
+import java.util.Set;
+
+/**
+ *
+ * @author Nikhil
+ */
+public class Memory
+{
+
+    public static HashMap memory;
+    public static Object[][] disp_mem;
+    public static Integer n;
+
+    public Memory()
+      {
+        memory = null;
+        memory = new HashMap();
+        n = new Integer(0);
+
+      }
+
+    public static int get(int address)
+      {
+        try
+        {// Integer data=new Integer(0);
+            Integer data = (Integer) memory.get(address);
+            return data;
+
+        } catch (Exception e)
+        {
+            FrontEnd.appendToPane(FrontEnd.statuswindow,"ERROR(" + ScanFile.current_line + "): Blank Memory Accesed Zero Is Being Retuned\n",Color.RED);
+//            frontend.FrontEnd.statuswindow.append("ERROR in line " + ScanFile.current_line + "Blank Memory Accesed zero is being retuned\n");
+            //frontend.FrontEnd.statuswindow.setCaretPosition(frontend.FrontEnd.statuswindow.getText().length());
+            frontend.FrontEnd.exceptionraised++;
+            return 0;
+        }
+      }
+
+    public static int getsize()
+      {
+        return memory.size();
+      }
+
+    public static void put(int b, int address)
+      {
+        memory.put(address, b);
+        n = memory.size();
+
+      }
+
+    public static Object[][] display(int ch)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.